internal/storage/postgres/treecluster/get.go

```diff
@@ -7,26 +7,55 @@ import (
 
 	"github.com/green-ecolution/green-ecolution-backend/internal/entities"
 	"github.com/green-ecolution/green-ecolution-backend/internal/logger"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage"
 	sqlc "github.com/green-ecolution/green-ecolution-backend/internal/storage/postgres/_sqlc"
 	"github.com/twpayne/go-geos"
 )
 
-func (r *TreeClusterRepository) GetAll(ctx context.Context) ([]*entities.TreeCluster, error) {
+func (r *TreeClusterRepository) GetAll(ctx context.Context) ([]*entities.TreeCluster, int64, error) {
 	log := logger.GetLogger(ctx)
-	rows, err := r.store.GetAllTreeClusters(ctx)
+	page, pageOk := ctx.Value("page").(int32)
+	limit, limitOK := ctx.Value("limit").(int32)
+
+	if !pageOk || !limitOK {
+		log.Debug("pagination values not found in context")
```

Comment:
Maybe print the raw values inside the context
```suggestion
		log.Debug("pagination values not found in context", "page", ctx.Value("page"), "limit", ctx.Value("limit")
```

Reply:
Sure, you can test it. If there is no value, the value is `nil`, otherwise an actual value which the logger will try to display as a string.

This line will produce this output:

```go
slog.Info("test to get not existing value in ctx", "ctx_value", ctx.Value("not_existing"))
```

```
[Jan 26 16:32:13] INFO main.go:84 test to get not existing value in ctx ctx_value="<nil>"
```